src/components/OAuth2Login/OAuth2Login.jsx

```diff
@@ -0,0 +1,156 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+import PopupWindow from './PopupWindow';
+import { toQuery } from './utils';
+
+const responseTypeLocationKeys = {
+  code: 'search',
+  token: 'hash',
+};
+
+const responseTypeDataKeys = {
+  code: 'code',
+  token: 'access_token',
+};
+
+class OAuth2Login extends Component {
+  constructor(props) {
+    super(props);
+    this.onBtnClick = this.onBtnClick.bind(this);
+    this.onRequest = this.onRequest.bind(this);
+    this.onSuccess = this.onSuccess.bind(this);
+    this.onFailure = this.onFailure.bind(this);
```

Comment:
We could potentially accomplish the same thing by using arrow functions, and then the constructor function could be scrapped.

```
onBtnClick = () => {
   const {
     butonText,
     ...
   }
}

...

const attrs = { onClick: this.onBtnClick };
```